test/e2e/apiextensions_compositions_test.go

```diff
@@ -412,3 +412,41 @@ func TestCircuitBreaker(t *testing.T) {
 			Feature(),
 	)
 }
+
+func TestCompositionExtraResources(t *testing.T) {
```

Comment:
we should consider using the "required resources" terminology that is now preferred, see https://github.com/crossplane/crossplane/pull/6686 for details on that.